packages/arb-token-bridge-ui/src/util/teleports/fetchTeleports.ts

```diff
@@ -0,0 +1,113 @@
+import { Provider } from '@ethersproject/providers'
+import {
+  fetchEthTeleportsFromSubgraph,
+  FetchEthTeleportsFromSubgraphResult
+} from './fetchEthTeleportsFromSubgraph'
+
+import {
+  fetchErc20TeleportsFromSubgraph,
+  FetchErc20TeleportsFromSubgraphResult
+} from './fetchErc20TeleportsFromSubgraph'
+import {
+  getL2ConfigForTeleport,
+  isTeleport
+} from '../../token-bridge-sdk/teleport'
+import { getChainIdFromProvider } from '../../token-bridge-sdk/utils'
+
+export type FetchTeleportsParams = {
+  sender?: string
+  receiver?: string
+  fromBlock?: number
+  toBlock?: number
+  parentChainProvider: Provider
+  childChainProvider: Provider
+  pageSize?: number
+  pageNumber?: number
+  searchString?: string
+}
+
+export type TeleportFromSubgraph =
+  | FetchEthTeleportsFromSubgraphResult
+  | FetchErc20TeleportsFromSubgraphResult
+
+export const fetchTeleports = async ({
```

Comment:
This is the singlular function which combines both the ETH and ERC20 teleport fetches. We use this in transaction-history instead of fetching ETH and ERC20 subgraphs individually and cluttering the code there.